api/v1beta2/awsmanagedcluster_types.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// AWSManagedClusterSpec defines the desired state of AWSManagedCluster
+type AWSManagedClusterSpec struct {
+	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// +optional
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+}
+
+// AWSManagedClusterStatus defines the observed state of AWSManagedCluster
+type AWSManagedClusterStatus struct {
+	// Ready is when the AWSManagedControlPlane has a API server URL.
+	// +optional
+	Ready bool `json:"ready,omitempty"`
+
+	// FailureDomains specifies a list fo available availability zones that can be used
+	// +optional
+	FailureDomains clusterv1.FailureDomains `json:"failureDomains,omitempty"`
+
+	// Conditions provide observations of the operational state of AWSManagedCluster.
+	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
```

Comment:
This is not a blocking comment considering that e2e test finally passed and the PR is ready to be merged. 

Considering AWSManagedCluster is a pass-through layer and there is no code that sets `Conditions`, I think `Conditions` field is unnecessary. At the same time, no harm leaving it in case we want to add Conditions in future.

Reply:
Conditions have been remove. We can always add them back in if we need them in the future.